src/dndbeyond/content-scripts/character.js

```diff
@@ -997,58 +998,124 @@ function rollAction(paneClass, force_to_hit_only = false, force_damages_only = f
             }
         }
     }
-    
-    //Cleric Blessed Strikes
-    if (character.hasClassFeature("Blessed Strikes") &&
-        character.getSetting("cleric-blessed-strikes", false) &&
-        spell_level.includes("Cantrip")) {
-        damages.push("1d8");
-        damage_types.push("Blessed Strikes");
+
+    if (character.hasClass("Ranger")) {
+        // Ranger: Favored Foe
+        if (character.hasClassFeature("Favored Foe") &&
+            character.getSetting("ranger-favored-foe", false) &&
+            to_hit !== null) {
+            const ranger_level = character.getClassLevel("Ranger");
+            damages.push(ranger_level < 6 ? "1d4" : ( ranger_level < 14 ? "1d6" : "1d8"));
+            damage_types.push("Favored Foe");
+        }
+        
+        // Ranger: Gathered Swarm
+        if (character.hasClassFeature("Gathered Swarm") &&
+            character.getSetting("ranger-gathered-swarm", false) &&
+            to_hit !== null) {
+            const ranger_level = character.getClassLevel("Ranger");
+            damages.push(ranger_level < 11 ? "1d6" : "1d8");
+            damage_types.push("Gathered Swarm");
+        }
     }
 
-    if (character.hasClassFeature("Enhanced Bond") &&
-        character.getSetting("wildfire-spirit-enhanced-bond", false) &&
-        damages.length > 0) {
-        for (let i = 0; i < damages.length; i++){
-            if (damage_types[i] === "Fire") {
-                damages.push("1d8");
-                damage_types.push("Enhanced Bond");
-                break;
-            }
+    if (character.hasClass("Warlock")) {
+        // Warlock: The Hexblade: Hexblade's Curse
+        if (damages.length > 0 &&
+            character.getSetting("warlock-hexblade-curse", false) &&
+            character.hasClassFeature("Hexblade’s Curse") &&
+            character._proficiency !== null) {
+            damages.push(character._proficiency);
+            damage_types.push("Hexblade's Curse");
+        }
+        
+        // Warlock: Genie Patron: Genie's Wrath
+        if (character.hasClassFeature("Genie’s Vessel") &&
+            character.getSetting("genies-vessel", false) && to_hit != null) {
+            damages.push(character._proficiency);
+            damage_types.push("Genie's Wrath");
+        }
+    }
+
+    if (character.hasClass("Wizard")) {
+        // Wizard: School of Evocation: Empowered Evocation
+        if (character.hasClassFeature("Empowered Evocation") &&
+            character.getSetting("empowered-evocation", false) &&
+            spell_level.includes("Evocation")) {
+            damages.push(`${parseInt(character.getAbility("INT").mod)}`);
+            damage_types.push("Empowered Evocation");
         }
     }
 
+    // Protector Aasimar: Radiant Soul Damage
+    if (character.hasRacialTrait("Radiant Soul") &&
+        character.getSetting("protector-aasimar-radiant-soul", false)) {
+        damages.push(character._level);
+        damage_types.push("Radiant Soul");
+    }
```

Comment:
Same for radiant soul. Basically you should just be calling `handleSpecialGeneralAttack` in the roll spell attack function so radiant soul, and hexblade's curse and the other stuff gets handled automatically. You should never have an ability handled in two of the functions.